extensions/s3/src/main/java/io/deephaven/extensions/s3/ProfileCredentials.java

```diff
@@ -0,0 +1,28 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.extensions.s3;
+
+import org.jetbrains.annotations.NotNull;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider;
+
+/**
+ * Profile specific credentials that uses a custom profile name or profile file.
+ *
+ * @see ProfileCredentialsProvider
+ */
+enum ProfileCredentials implements AwsSdkV2Credentials {
+    INSTANCE;
+
+    @Override
+    public final AwsCredentialsProvider awsV2CredentialsProvider(@NotNull final S3Instructions instructions) {
+        if (instructions.profileName().isEmpty() && instructions.aggregatedProfileFile().isEmpty()) {
+            throw new IllegalArgumentException("Profile name or profile file must be provided for profile credentials");
```

Comment:
I don't think this should be an error case; the user might want to force the default profile, `software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider#create()`.